File: ployglots/mta-ployglot-pkg-figma-api/packages/py/figma_api/resources/variables.py
"""Variables — Figma REST API coverage.

Tier: enterprise only. Returns 403 without an Enterprise/Org plan.
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any
from urllib.parse import quote

if TYPE_CHECKING:
    from .._figma_client import FigmaClient


class VariablesResource:
    def __init__(self, client: FigmaClient) -> None:
        self._client = client

    async def list_local(self, file_key: str) -> dict[str, Any]:
        """GET /v1/files/:file_key/variables/local — local variables."""
        return await self._client.get(
            f"/v1/files/{quote(file_key, safe='')}/variables/local"
        )

    async def list_published(self, file_key: str) -> dict[str, Any]:
        """GET /v1/files/:file_key/variables/published — published variables."""
        return await self._client.get(
            f"/v1/files/{quote(file_key, safe='')}/variables/published"
        )

    async def post_variables(self, file_key: str, body: dict[str, Any]) -> dict[str, Any]:
        """POST /v1/files/:file_key/variables — upsert variables / collections / modes."""
        return await self._client.post(
            f"/v1/files/{quote(file_key, safe='')}/variables", json=body
        )

File: ployglots/mta-ployglot-pkg-figma-api/packages/ts/src/resources/variables.ts
// @ts-nocheck
/**
 * Variables — Figma REST API coverage.
 *
 * Tier: **enterprise only**. These endpoints return 403 without an
 * Enterprise/Org plan. `FigmaAuthError` surfaces that — read the
 * Figma docs for plan requirements.
 */

import type { FigmaClient } from '../figma-client.js';

export interface VariablesPublishRequest {
  variableCollections?: unknown[];
  variableModes?: unknown[];
  variables?: unknown[];
  variableModeValues?: unknown[];
}

export class VariablesResource {
  constructor(private readonly client: FigmaClient) {}

  /** GET /v1/files/:file_key/variables/local — locally-defined variables. */
  async listLocal(fileKey: string): Promise<unknown> {
    return this.client.get(`/v1/files/${encodeURIComponent(fileKey)}/variables/local`);
  }

  /** GET /v1/files/:file_key/variables/published — published variables. */
  async listPublished(fileKey: string): Promise<unknown> {
    return this.client.get(`/v1/files/${encodeURIComponent(fileKey)}/variables/published`);
  }

  /**
   * POST /v1/files/:file_key/variables — upsert a batch of variables /
   * collections / modes. See Figma's variables API docs for the full
   * action shape (create / update / delete under one envelope).
   */
  async postVariables(fileKey: string, body: VariablesPublishRequest): Promise<unknown> {
    return this.client.post(`/v1/files/${encodeURIComponent(fileKey)}/variables`, { body });
  }
}
